src/main/java/org/rust/lang/core/lexer/LexerBaseKt.java

```diff
@@ -0,0 +1,61 @@
+package org.rust.lang.core.lexer;
+
+import com.intellij.lexer.LexerBase;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Small utility class to ease implementing {@link LexerBase} in Kotlin.
+ */
+public abstract class LexerBaseKt extends LexerBase {
+    private int state;
+    private int tokenStart;
+    private int tokenEnd;
+    private CharSequence bufferSequence;
+    private int bufferEnd;
+
+    @Override
+    public int getState() {
+        return state;
+    }
+
+    protected void setState(int state) {
+        this.state = state;
+    }
+
+    @Override
+    public int getTokenStart() {
+        return tokenStart;
+    }
+
+    protected void setTokenStart(int tokenStart) {
+        this.tokenStart = tokenStart;
+    }
+
+    @Override
+    public int getTokenEnd() {
+        return tokenEnd;
+    }
+
+    protected void setTokenEnd(int tokenEnd) {
+        this.tokenEnd = tokenEnd;
+    }
+
+    @NotNull
+    @Override
+    public CharSequence getBufferSequence() {
+        return bufferSequence;
+    }
+
+    protected void setBufferSequence(CharSequence bufferSequence) {
+        this.bufferSequence = bufferSequence;
+    }
+
+    @Override
+    public int getBufferEnd() {
+        return bufferEnd;
+    }
+
+    protected void setBufferEnd(int bufferEnd) {
+        this.bufferEnd = bufferEnd;
+    }
+}
```

Comment:
This seems to work for me

```Kotlin
package org.rust.lang.core.lexer

import com.intellij.lexer.LexerBase

abstract class LexerBaseKt : LexerBase() {
    protected var myState: Int = 0
    protected var myTokenStart: Int = 0
    protected var myTokenEnd: Int = 0
    protected lateinit var myBufferSequence: CharSequence
    protected var myBufferEnd: Int = 0

    override fun getState(): Int = myState

    override fun getTokenStart(): Int = myTokenStart

    override fun getTokenEnd(): Int = myTokenEnd

    override fun getBufferSequence(): CharSequence = myBufferSequence

    override fun getBufferEnd(): Int = myBufferEnd
}
```

This `my` prefixes are not exactly pretty, but it is a common convention in IDEA anyway.